sklearn/ensemble/_hist_gradient_boosting/binning.py

```diff
@@ -17,6 +18,52 @@
 from ._bitset import set_bitset_memoryview
 
 
+def _pack_matrix(sparse_matrix):
```

Comment:
We can convert into a csc feeding it into `_BinMapper` and then build the bins from there.